### ISSUE TITLE
Label text paste on play inconsistency [mod-script-pipe]

### ISSUE THREAD

**Issue by idlesign** [audacity/audacity#4751]
### Bug description

Nothing is pasted from the clipboard into a label on play when Audacity is launched from terminal.
When launched from applications or dock it works as expected.

### Steps to reproduce

1. Run from terminal: `$ audacity`
2. Start recording
3. Copy text to clipboard
4. Issue `AddLabelPlaying` and `Paste` commands (or `PasteNewLabel`)

### Expected behavior

Text from clipboard should be pasted into new label.

### Actual behavior

New label is created and set active.
Yet no text is pasted.

If stopped text pastes just fine.

### Audacity Version

stable version (from a Linux repository)

### Operating system

Linux

### Additional context

Ubuntu 22.04
Wayland
Audacity 2.4.2

**Comment by Ryan-Miller-Audacity** [comment]
Testing this in Audacity 3.3.2 Appimage on Linux Fedora I don't see a problem in behavior here. The paste option is disabled when recording audio (you can see this by its gray state from the edit dropdown menu) and the "Paste text to new label" option within Edit->Labels is also disabled while recording. The paste text to new label option works as expected when not recording and labels can still be created/typed within when recording as well. 

Could you provide some more details on this issue? Perhaps a video of the behavior you're describing? From what I see things are working as expected. If you are using Audacity 2.4.2 that is a _very_ old release. Can you see if you have this issue on Audacity 3.3.2 (current public release) as well?

**Comment by idlesign** [comment]
Thank you for the response.

> Could you provide some more details on this issue? 

I'm trying to investigate the reasons using **another computer with a similar setup** (Ubuntu 22.04, Wayland, Audacity 2.4.2).
But so far without a luck: the second part of [the video](https://disk.yandex.com/i/czqhxWbTC1boFw) shows it (`some` text is sucessfully pasted into labels).

I've made a script to facitilate a reproduction — [here](https://gist.github.com/idlesign/23797adae3a813f85718cea12e708fd1).
* Run Audacity with pipe enabled from programs
* Run `$ python checker.py` (python 3.6+ is required)

Record is started. Empty labels are added (if no text in clipboard so far). 
Now you may copy some text into the clipboard (e.g. from gnote, gedit, etc). Labels are filled during the record with the contents of your clipboard, every 1.5 sec. That's expected.

What've seen on **the first computer** are empty labels when Audacity has been started from a terminal.
I'll try the script and make another video as soon as as reach the first computer.

> Can you see if you have this issue on Audacity 3.3.2 (current public release) as well?

There's a problem with pipe in appImage as mentioned here: https://github.com/audacity/audacity/issues/3324#issuecomment-1276285843

Yet I'll also try to check 3.3.2 somehow, and come back later.

**Comment by Ryan-Miller-Audacity** [comment]
Thanks for the additional details. Best course of action for now it seems will be for you to continue investigating between the two machines and also see if you're able to reproduce it within Audacity 3.3.2 and then we can move forward from there.

**Comment by idlesign** [comment]
So I was able to reproduce in on the first machine yesterday. 
The previous description needs a correction though: `Yet no text is pasted` should become `The text that was copied into the clipboard before Audacity start is pasted. A text copied after the start is not pasted — the old value is pasted instead`.

Now for the weirdest part. 
To capture the above mentioned behaviour I've installed OBS. OBS in its turn required from me reinstall of `gnome-control-center` and switching from `nouveau` to `nvidia` drivers.
But after that's been done, I was not able to reproduce anything similar. 
I suppose the beviour may somehow be related to the desktop manager on a multimonitor setup, but that's just a guess.

With that I want to apologize for the disturbance and close the issue.

----

By the way, I've tested both 2.4.2 and 3.3.2 (see https://github.com/audacity/audacity/issues/3324#issuecomment-1572021711 for a hint on how to make an appImage work) and both are good.

Wish you luck. 
Thank you for Audacity.
